### DISCUSSION TITLE
How get real time data on custom route that I have created?

### DISCUSSION THREAD

**Discussion by samimhamza** [pocketbase/pocketbase#2376] [Q&A] [answered]
I have to get recommended jobs for each user according to skills similarities. and for this I have to create custom route. so how can I get real time subscription for recommended job and custom route that I have made?

**Comment by ganigeorgiev** [comment]
I'm not sure I follow. The realtime events are fired on collection record changes (create/update/delete) and as long as you are updating the record using the `app.Dao()` in your custom route, you should be able to subscribe to the collection and receive the related action event.

Could you elaborate or provide a code sample with your custom route?

**Reply by samimhamza** [reply]
yes real time events fired on collections. how should I filter real time events for example only get real time changes on recommended jobs or on jobs that has been added in last week or something like that?

**Reply by ganigeorgiev** [reply, marked as answer]
Without a code sample how a "recommended job" is identified, I'm not sure how to help.

The general approach would be to subscribe to the `jobs` collection and inside the callback to call your _/recommended-jobs_ custom route so that you can update your local state.

**Reply by samimhamza** [reply]
I didn't implement the code yet. I was just thinking about how to implement real time on filtered data of a collection and that sounds a good solution, thank you
